src/NATS.Client.Core/NatsPooledConnection.cs

```diff
@@ -0,0 +1,13 @@
+namespace NATS.Client.Core;
+
+internal class NatsPooledConnection : NatsConnection
```

Comment:
This should be `sealed`. Alternatively, `NatsConnection` could be sealed and this could be a readonly struct, e.g.

```
readonly struct PooledNatsConnection : INatsConnection {
  private readonly NatsConnection _conn;
  // delegate all calls to _conn, except for DisposeAsync
}
```

Reply:
@jasper-d this can't be a struct, as it needs to inherit from NatsConnection, not the interface in order to prevent disposing of a pooled connection.